source/game_sa/ClothesBuilder.cpp

```diff
@@ -75,34 +91,169 @@ void CClothesBuilder::ReleaseGeometry(int32 numToRelease) {
         --i;
 }
 
-// 0x5A47E0
-void FindAtomicFromNameCB(RpAtomic* atomic, void* data) {
-    plugin::Call<0x5A47E0, RpAtomic*, void*>(atomic, data);
-}
-
 // 0x5A4810
-void GetAtomicWithName(RpClump* clump, const char* name) {
-    plugin::Call<0x5A4810, RpClump*, const char*>(clump, name);
+RpAtomic* GetAtomicWithName(RpClump* clump, const char* name) {
+    struct Context {
+        notsa::ci_string_view name{};
+        RpAtomic*             a{};
+    } c{name};
+    RpClumpForAllAtomics(clump, [](RpAtomic* a, void* data) { // 0x5A47E0
+        auto& c = *static_cast<Context*>(data);
+
+        if (c.name == GetFrameNodeName(RpAtomicGetFrame(a))) {
+            c.a = a;
+        }
+
+        return a;
+    }, &c);
+    return c.a;
```

Comment:
```suggestion
    struct Context {
        notsa::ci_string_view name{};
        RpAtomic*             atomic{};
    } c{name};
    RpClumpForAllAtomics(clump, [](RpAtomic* a, void* data) { // 0x5A47E0
        auto& c = *static_cast<Context*>(data);

        if (c.name == GetFrameNodeName(RpAtomicGetFrame(a))) {
            c.atomic = a;
        }

        return a;
    }, &c);
    return c.atomic;
```

`c.a = a` then `return a` is ugly and confusing imo